lib/ZMQClient/methods/connect.js

```diff
@@ -26,50 +27,59 @@ async function connect(connectionOptions = {}) {
   const socket = new Subscriber(socketOptions);
   this.socket = socket;
 
-  return new Promise(async (resolve, reject) => {
-    // We only return socket when we actually established a connection
-    function onConnectionListener() {
-      self.isConnected = true;
-      resolve(socket);
-    }
-    function onEndListener() {
-      self.isConnected = false;
-      resolve(socket);
+  function onEndListener() {
```

Comment:
If you use arrow function definitions then you can use `this` inside and don't need `self` variable.